### ISSUE TITLE
Attributes set on the svg element are not inherited.

### ISSUE THREAD

**Issue by WarWithinMe** [adobe/svg-native-viewer#114]
In the following svg document, the red path should be rendered as a border surrounding the green rectangle. However, svg-native-viewer will render it as a red rectangle. "fill-rule" defined in svg element is not inherited by the children.

```
<svg width="100%" height="100%" viewBox="0 0 100 100" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" fill-rule="evenodd" clip-rule="evenodd">
    <g transform="matrix(1.21071,0,0,1.20782,1.32858,1.58746)">
       <rect x="2" y="8" width="76.402" height="64.165" fill="green"/>
       <path d="M78.402,8L2,8L2,72.165L78.402,72.165L78.402,8ZM4.065,10.07L4.065,70.095L76.337,70.095L76.337,10.07L4.065,10.07Z" fill="red"/>
    </g>
</svg>
```

Expected:
![Expected](https://user-images.githubusercontent.com/567110/89355982-474d5680-d6ef-11ea-9674-1da362c2e6aa.jpg)

Output:
![Output](https://user-images.githubusercontent.com/567110/89355984-487e8380-d6ef-11ea-8bf9-68e618266060.jpg)

I think it's because svg node is bypassed during traversing:
https://github.com/adobe/svg-native-viewer/blob/1901c27b85e66809144e94b4de86f5b181158b1e/svgnative/src/SVGDocumentImpl.cpp#L98

**Comment by mpsuzuki** [comment]
Thank you for reporting! Please let me spend a few days to check this...

**Comment by mpsuzuki** [comment]
@WarWithinMe , thank you for reporting this, your titling of this issue is very helpful. I confirmed the issue, and the appearance is fixed if I insert the fill-rule to each elements.
In SVG Native Viewer, the inheritance of some attributes is implemented as stacks (please find mStrokeStyleStack, mFillStyleStack, and mGroupStack in SVGDocumentImpl class).
At present, no stack to store fill-rule or clip-rule. Maybe I should add them to resolve this issue.

**Comment by WarWithinMe** [comment]
Aren't fill-rule and clip-rule stored in `FillStyle`?

I think the root cause is that in `void SVGDocumentImpl::TraverseSVGTree(XMLNode* rootNode)`, the svg's properties are completely ignored (not pushed on to the stacks). I guess other properties like fill-color set on the SVG node are also not inherited.

Currently my workaround is to patch SVGDocumentImpl:
L98:
```
ParseChildren(rootNode);
```
to:
```
ParseChild(rootNode);
```

And in `void SVGDocumentImpl::ParseChild(XMLNode* child)`, I added a condition for svg node:
```
    else if (!strcmp(elementName, kSvgElem))
    {
        mFillStyleStack.push(fillStyle);
        mStrokeStyleStack.push(strokeStyle);

        ParseChildren(child);

        mFillStyleStack.pop();
        mStrokeStyleStack.pop();
    }
```

**Comment by mpsuzuki** [comment]
>Aren't fill-rule and clip-rule stored in FillStyle?
Indeed, sorry for overlooking that. Also thanks to your proposed patch.

**Comment by mpsuzuki** [comment]
Just confirmed your patch fix this issue, thanks again! Please let me spend some time to understand the idea of the original author why ParseChildren() was used in this case.

**Comment by dirkschulze** [comment]
@mpsuzuki Did you have a chance to see if this was an oversight? Didn't check myself yet.

**Comment by dirkschulze** [comment]
I see why it was done: `ParseChild` is called by `ParseChildren`. In SVG Native, no descendant is allowed to be an `<svg>` element. This probably was an easy way to deal with that but didn't take inheritance into account.

**Comment by dirkschulze** [comment]
Fixed by #122
